### ISSUE TITLE
Token Discovery Service API Differences between documentation and reality

### ISSUE THREAD

**Issue by jangelfdez** [docker/swarm#569]
Hi!

I'm playing with the Token discovery service API following the API defined here: https://github.com/docker/swarm/blob/master/discovery/token/README.md

* Create a new cluster: OK
* Add new nodes to a cluster: OK
* List nodes in a cluster: 

Expected:  ["<ip:port1>", "<ip:port2>"] 
Received: 
[
"------WebKitFormBoundaryFHknzaNRPP4Fz8MW\ \ Content-Disposition: form-data; name=\"data\"\ \ \ \ ip.:port1\ \ ------WebKitFormBoundaryFHknzaNRPP4Fz8MW--\ \ ",
"------WebKitFormBoundaryF32dMRrMdcLAGsij\ \ Content-Disposition: form-data; name=\"data\"\ \ \ \ ip:port2 \ ------WebKitFormBoundaryF32dMRrMdcLAGsij--\ \ "
]

* Delete a cluster (all the nodes in a cluster)

Expected: OK
Received: 
pq: update or delete on table "clusters" violates foreign key constraint "nodes_cluster_id_fkey" on table "nodes"

If anyone want to try, I have created a <a href="http://www.joseangelfernandez.es/blog/wp-content/uploads/2015/04/Token - Docker Hub Discovery Service.json.postman_collection">Postman collection </a> and <a href=""http://www.joseangelfernandez.es/blog/wp-content/uploads/2015/04/Token%20-%20Docker%20Hub%20Discovery%20Service%20Environment.postman_environment">environment file </a>

**Comment by vieux** [comment]
@jangelfdez Hi we are aware of the delete issue we are going to fix it soon.

Regarding the `List nodes in a cluster` it's probably an error from you, because `swarm list` is using it and it works.

**Comment by vieux** [comment]
@jangelfdez to add a node you just need <ip>:<port> in the post body. you shouldn't use a <data> key with <ip>:<port> value

**Comment by jangelfdez** [comment]
Hi Vieux,

calling it from different tools and computers produces the same output for me:

*Web browser (Chrome, IE): 
```
["------WebKitFormBoundaryFHknzaNRPP4Fz8MW\r\nContent-Disposition: form-data; name=\"data\"\r\n\r\n192.168.1.2:2375\r\n------WebKitFormBoundaryFHknzaNRPP4Fz8MW--\r\n","------WebKitFormBoundaryF32dMRrMdcLAGsij\r\nContent-Disposition: form-data; name=\"data\"\r\n\r\n192.168.1.3:2375\r\n------WebKitFormBoundaryF32dMRrMdcLAGsij--\r\n"]
```

* PowerShell: 
```
Invoke-RestMethod -Method GET -Uri https://discovery-stage.hub.docker.com/v1/clusters/c0fc6b041cac5a7aeaeecXXXX
------WebKitFormBoundaryFHknzaNRPP4Fz8MW
Content-Disposition: form-data; name="data"
192.168.1.2:2375
------WebKitFormBoundaryFHknzaNRPP4Fz8MW--
------WebKitFormBoundaryF32dMRrMdcLAGsij
Content-Disposition: form-data; name="data"
192.168.1.3:2375
------WebKitFormBoundaryF32dMRrMdcLAGsij--
```

* Curl & wget
 ```
["------WebKitFormBoundaryFHknzaNRPP4Fz8MW\r\nContent-Disposition: form-data; name=\"data\"\r\n\r\n192.168.1.2:2375\r\n------WebKitFormBoundaryFHknzaNRPP4Fz8MW--\r\n","------WebKitFormBoundaryF32dMRrMdcLAGsij\r\nContent-Disposition: form-data; name=\"data\"\r\n\r\n192.168.1.3:2375\r\n------WebKitFormBoundaryF32dMRrMdcLAGsij--\r\n"]
```

Should it be normal? Running from the same computers a Swarm List, I get also the correct values on the screen.

**Comment by jangelfdez** [comment]
@vieux thanks, after reading in the documentation 
```
(data="<ip:port2>") 
```
i thought it was a key=value pair. Trying with the data only in the body works OK ;)

**Comment by vieux** [comment]
@jangelfdez feel free to do a PR on the documentation if you have a better way to explain it.

**Comment by jangelfdez** [comment]
Solved #570
